dspy/evaluate/evaluate.py

```diff
@@ -227,7 +227,8 @@ def wrapped_program(example_idx, example):
             results = [(example, prediction, score) for _, example, prediction, score in predicted_devset]
 
         data = [
-            merge_dicts(example, prediction) | {"correct": score} for _, example, prediction, score in predicted_devset
+            dict(example) | {"prediction": prediction, "correct": score}
```

Comment:
Thanks so much @dbczumar ! Everything looks great to me, besides the following questions.

Is the moving of `prediction` under a key called "prediction" intentional? What if the prediction has many fields? How will that get displayed? I imagine it breaks the niceness of columns, unless the table displayer is smart and shows some kind of sub-columns.

I do like getting rid of `merge_dicts` which is not needed in 3.9+.

Reply:
@okhat Got it - makes perfect sense :). For dict-like outputs, I've restored the preexsting behavior. For non-dict-like outputs, I think including the full value in a `predictions` column makes sense.